timemachine/fe/mcgregor.py

```diff
@@ -358,28 +360,33 @@ def mcs(
             filter_fxn,
         )
 
-        total_nodes_visited += mcs_result.nodes_visited
+        visits_left -= mcs_result.nodes_visited
 
-        # If timed out, either due to max_visits or max_cores, raise exception.
-        if mcs_result.timed_out:
-            warnings.warn(
-                f"Reached max number of visits/cores: {len(mcs_result.all_maps)} cores with {mcs_result.nodes_visited} nodes visited. "
-                "Cores may be suboptimal.",
-                MaxVisitsWarning,
-            )
+        total_nodes_visited += mcs_result.nodes_visited
 
         if len(mcs_result.all_maps) > 0:
+            # If we timed out but got cores, throw a warning
+            if mcs_result.timed_out:
+                warnings.warn(
+                    f"Reached max number of visits/cores: {len(mcs_result.all_maps)} cores with {mcs_result.nodes_visited} nodes visited. "
+                    "Cores may be suboptimal.",
```

Comment:
Suboptimal implies something about the quality of the core that I don't think we can imply at this stage.

If you have at least one core that's been found, but you timed out, the core(s) you did find are probably still comparable to the hidden, remaining cores that you didn't find. Maybe to be more precise:

``` python
if mcs_result.timed_out and len(mcs_result.all_maps) < max_cores:
    warnings.warn(f"Inexhaustive search: reached max number of visits ({mcs_result.nodes_visited}) and found only {len(mcs_result.all_maps)} out of {max_cores} desired cores."
```

Reply:
Adjusted in https://github.com/proteneer/timemachine/pull/1357/commits/5ae26fa556061bcf280716949f4e740783228b05 and https://github.com/proteneer/timemachine/pull/1357/commits/f6ec329025918bce2a86b5f8ae3a3c6c0fc20bac to be more precise